lib/extract-templates.js

```diff
@@ -0,0 +1,133 @@
+import { parseTemplates } from 'ember-template-imports/lib/parse-templates.js';
+
+export const SUPPORTED_EXTENSIONS = ['js', 'ts', 'gjs', 'gts'];
```

Comment:
I wonder if there is a way to have these be opt-in? :thinking: 

I've been curious if these supported extensions will have an impact on an existing codebase.

Any chance, you can look in to that, @ventuno?

Ideally, with nothing but default configs, js/ts (tests, for example) remain unlinted. -- we can flip the default in a major tho

Reply:
Thanks @NullVoxPopuli, the only way I can think of doing this, is by using a command line argument which would enable parsing for script files. Thoughts?

Reply:
The way ESLint etc. approach this is allowing you to specify what to include in your config, so that's probably the forward-and-backward-compatible way to do it here: instead of hard-coding all the extensions, introduce a config option which specifies extensions to tackle, supplies a default of just `.hbs`, and then allows folks to opt in new extensions as well, with an explanation of how those will be handled. (Perhaps we need two things: template-only file extensions and embedded template extensions?)

Reply:
`ember-template-lint` has the [`ignore-pattern`](https://github.com/ember-template-lint/ember-template-lint/blob/master/lib/helpers/cli.js#L104) argument which could be used to this effect.
What about adding to the list of default values all patterns we want to exclude?
Something like:
```
'ignore-pattern': {
  describe: 'Specify custom ignore pattern (can be disabled with --no-ignore-pattern)',
  type: 'array',
  default: [..., '**/*.js', '**/*.gjs', ...],
},
```
provided that works as expected.

Reply:
@NullVoxPopuli, @chriskrycho so I just updated the code to ignore script files by default and added tests. Let me know what you think, but this should be the most straightforward way to do this.